allennlp/models/semantic_parsing/nlvr/nlvr_coverage_semantic_parser.py

```diff
@@ -108,7 +109,11 @@ def __init__(self,
             self._dynamic_cost_rate = dynamic_cost_weight["rate"]
         self._penalize_non_agenda_actions = penalize_non_agenda_actions
         self._last_epoch_in_forward: int = None
-        if initial_mml_model_file is not None:
+        # TODO (pradeep): Checking whether file exists here to avoid raising an error when we've
```

Comment:
Yes, we don't need the MML model we used to initialize the ERM model when we using the trained ERM model to decode. I cannot think of a way to distinguish between training a decoding in the constructor. I guess I can pass yet another argument to the constructor, but that seems unnecessary.

Reply:
This is the right thing to do, I think, except maybe log a warning if the file isn't present (in case you're actually training and are expecting the file to be there).  Fixing this issue was part of the original motivation of the `files_to_archive` stuff that @joelgrus added a while back, though that work didn't get as far as adding a mechanism for removing keys from a config that are only necessary at training time.  I'll open an issue to track this.